File: htp-hotel-vendor-1-backend/src/Akros.Htp.Vendor.Backend.DataAccess.Model/Image/ImageEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Akros.Htp.Vendor.Backend.DataAccess.Model.Hotel;

namespace Akros.Htp.Vendor.Backend.DataAccess.Model.Image
{
    public class ImageEntity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public long Id { get; set; }

        [Required]
        public string Title { get; set; }
        
        public string AlternativeText { get; set; }

        public int Width { get; set; }
        
        public int Height { get; set; }
        
        public byte Image { get; set; }
        
        public byte ThumbImage { get; set; }

        public string ImageUrl { get; set; }

        public string ThumbImageUrl { get; set; }

        public HotelEntity Hotel { get; set; }

        public long HotelId { get; set; }

    }
}

File: htp-hotel-vendor-2-backend/src/main/java/ch/akros/htp/hotelvendor/entity/image/ImageEntity.java
package ch.akros.htp.hotelvendor.entity.image;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

import ch.akros.htp.hotelvendor.entity.hotel.HotelEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name = "image")
public class ImageEntity {
	@Id
	@Column(name = "id", nullable = false, unique = true)
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@NotBlank
	@Size(min = 3, max = 120)
	@Column(name = "NAME")
	private String name;

	@Column(name = "WIDTH")
	private int width;

	@Column(name = "HEIGHT")
	private int height;

	@Column(name = "DOWNLOAD_URL")
	private String downloadUrl;

	@ManyToOne
	@JoinColumn(referencedColumnName = "id", name = "hotel_id", foreignKey = @ForeignKey(name = "hotel_id_image_fk"))
	private HotelEntity hotel;
}
